src/OpenLoco/Ui/ViewportInteraction.cpp

```diff
@@ -428,20 +895,82 @@ namespace OpenLoco::Ui::ViewportInteraction
+        auto& interaction = res.first;
+        switch (interaction.type)
+        {
+            case InteractionItem::track:
+                hasInteraction = rightOverTrack(interaction);
+                break;
+            case InteractionItem::entity:
+                // No interaction
+                break;
+            case InteractionItem::trackExtra:
+                hasInteraction = rightOverTrackExtra(interaction);
+                break;
+            case InteractionItem::signal:
+                hasInteraction = rightOverSignal(interaction);
+                break;
+            case InteractionItem::trackStation:
+                hasInteraction = rightOverTrackStation(interaction);
+                break;
+            case InteractionItem::roadStation:
+                hasInteraction = rightOverRoadStation(interaction);
+                break;
+            case InteractionItem::airport:
+                hasInteraction = rightOverAirport(interaction);
+                break;
+            case InteractionItem::dock:
+                hasInteraction = rightOverDock(interaction);
+                break;
+            case InteractionItem::road:
+                hasInteraction = rightOverRoad(interaction);
+                break;
+            case InteractionItem::roadExtra:
+                hasInteraction = rightOverRoadExtra(interaction);
+                break;
+            default:
+                if (Input::hasFlag(Input::Flags::toolActive) || Input::hasFlag(Input::Flags::flag6))
+                {
+                    if (WindowManager::find(WindowType::construction) == nullptr)
+                    {
+                        if (interaction.type == InteractionItem::building)
+                        {
+                            hasInteraction = rightOverBuilding(interaction);
+                        }
+                        break;
+                    }
+                }
+                // 0x4CDBC5
+                switch (interaction.type)
+                {
+                    case InteractionItem::tree:
+                        hasInteraction = rightOverTree(interaction);
+                        break;
+                    case InteractionItem::wall:
+                        hasInteraction = rightOverWall(interaction);
+                        break;
+                    case InteractionItem::building:
+                        hasInteraction = rightOverBuildingConstruct(interaction);
+                        break;
+                    case InteractionItem::headquarterBuilding:
+                        hasInteraction = rightOverHeadquarters(interaction);
+                        break;
+                    default:
+                        hasInteraction = true;
+                        break;
+                }
+                break;
+        }
+        if (hasInteraction)
+        {
+            return interaction;
+        }
+        else
+        {
+            return InteractionArg{};
+        }
```

Comment:
This is down to personal preference, but if you like:
```suggestion
        return hasInteraction ? interaction : InteractionArg{};
```